File: 1. Two Sum/Solution.py
def twoSum(nums, target):
    hash_table = {}
    for i in range(len(nums)):
        complement = target - nums[i]
        if complement not in hash_table:
            hash_table[nums[i]] = i
        else:
            return [hash_table[complement], i]


print(twoSum([3, 2, 4], 6))

File: 1. Two Sum/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> index;
        
        for (int i=0; i<nums.size(); i++)
        {
            int compensate = target - nums[i];
            if (index.find(compensate) != index.end())
            {
                return {i, index.find(compensate)->second};
            }
            index.insert(pair<int,int>(nums[i],i));
        }
        return {};
    }
};